opendevin/controller/agent_controller.py

```diff
@@ -239,7 +237,10 @@ async def step(self, i: int) -> bool:
                 obs: Observation = AgentDelegateObservation(content='', outputs=outputs)
                 await self.add_history(NullAction(), obs)
                 self.delegate = None
-                self.delegateAction = None
+            elif self.delegate._is_stuck():
+                logger.info('Loop detected in DelegateAgent, stopping task')
+                await self.delegate.set_agent_state_to(AgentState.ERROR)
+                raise Exception('Loop detected, stopping task.')
             return False
```

Comment:
I _think_ we can just put
```
if self.delegate and self.delegate._is_stuck():
  return True
```
inside `_is_stuck`, and remove this part entirely. That would be ideal at least.